File: C++/longestSubstring.cpp
#include <string>
#include <unordered_set>
#include <algorithm>

class Solution {
public:
    int lengthOfLongestSubstring(std::string s) {
        int maxLength = 0;
        int left = 0, right = 0;
        std::unordered_set<char> seenChars;
        
        while (right < s.length()) {
            if (seenChars.find(s[right]) == seenChars.end()) {
                seenChars.insert(s[right]);
                maxLength = std::max(maxLength, right - left + 1);
                right++;
            } else {
                seenChars.erase(s[left]);
                left++;
            }
        }
        
        return maxLength;
    }
};

File: Java/longestSubstring.java
import java.util.HashSet;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxLength = 0;
        int left = 0, right = 0;
        HashSet<Character> seenChars = new HashSet<>();
        
        while (right < s.length()) {
            if (!seenChars.contains(s.charAt(right))) {
                seenChars.add(s.charAt(right));
                maxLength = Math.max(maxLength, right - left + 1);
                right++;
            } else {
                seenChars.remove(s.charAt(left));
                left++;
            }
        }
        
        return maxLength;
    }
    
    public static void main(String[] args) {
        Solution solution = new Solution();
        
        // Test cases
        System.out.println(solution.lengthOfLongestSubstring("abcabcbb")); // Output: 3
        System.out.println(solution.lengthOfLongestSubstring("bbbbb"));    // Output: 1
        System.out.println(solution.lengthOfLongestSubstring("pwwkew"));   // Output: 3
    }
}
